src/useCreateComponent.tsx

```diff
@@ -12,14 +11,18 @@ export const useCreateComponent = (
   const wrapper = <div ref={wrapperDivRef}></div>;
 
   React.useLayoutEffect(() => {
-    if (wrapperDivRef.current !== null && component === null) {
+    if (
```

Comment:
There is potential for a perf optimization here around loading of embedded components. Specifically, ConnectJS won't dynamically load the code for an embedded component until it is created.

This approach does not improve performance in that sense, since the component won't get created until that promise resolves. It only makes initializing with react a bit easier - I'd like to pursue an approach that does improve performance here.